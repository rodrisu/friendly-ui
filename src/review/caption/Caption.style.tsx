import styled from 'styled-components'

import { Button } from '../../button'
import { normalizeHorizontally } from '../../layout/layoutNormalizer'

export const StyledCaption = styled.div`
  display: flex;
  ${normalizeHorizontally};

  > * {
    flex: 1 0 auto;
  }

  > * + * {
    text-align: right;
  }

  ${Button} {
    display: inline;
  }
`
