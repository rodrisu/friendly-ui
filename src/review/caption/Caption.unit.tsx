import React from 'react'

import { fireEvent, render, screen } from '@testing-library/react'

import { Caption } from './index'

describe('Review/Caption', () => {
  it('should render the date', () => {
    render(<Caption isoDate="2017-08-07T14:10:40.526Z">08th August 2017</Caption>)

    expect(screen.getByText('08th August 2017')).toBeInTheDocument()
  })

  it('should render a link', () => {
    const onClick = jest.fn()
    render(
      <Caption
        isoDate="2017-08-07T14:10:40.526Z"
        secondaryText="Reply"
        href="https://www.blablacar.fr"
        onClick={onClick}
      >
        08th August 2017
      </Caption>,
    )

    expect(screen.getByRole('link', { name: 'Reply' })).toBeInTheDocument()
    fireEvent.click(screen.getByRole('link', { name: 'Reply' }))
    expect(onClick).toHaveBeenCalled()
  })
})
