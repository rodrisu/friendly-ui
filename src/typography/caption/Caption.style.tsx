import styled from 'styled-components'

import { color, font, fontWeight } from '../../_utils/branding'
import { Text } from '../index'

export const TextCaption = styled(Text)`
  color: ${color.lightMidnightGreen};
  font-size: ${font.s.size};
  font-weight: ${fontWeight.regular};
  line-height: ${font.s.lineHeight};
`
