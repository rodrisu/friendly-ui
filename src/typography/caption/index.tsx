export { TextCaption } from './Caption.style'
