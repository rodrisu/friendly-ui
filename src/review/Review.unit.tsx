import React, { Fragment } from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import { Caption } from './caption'
import { Review } from './index'

const defaultReviewProps = {
  title: 'Review title',
  text: 'Review content',
  formattedDatetime: '05 jul - 17:39',
  isoDatetime: '2017-08-07T14:10:40.526Z',
}

function assertReview(
  review: any,
  expectedTitle: string,
  expectedText: string,
  expectedDate: string,
  isReviewResponse: boolean,
) {
  const title = review.find('h2')
  expect(title.text()).toBe('Review title')
  const paragraph = review.find('p')
  expect(paragraph.text()).toBe('Review content')
  const date = review.find(Caption)
  expect(date.text()).toBe('05 jul - 17:39')
  const reviewResponse = review.find('.kirk-is-review-response')
  expect(reviewResponse.exists()).toBe(isReviewResponse)
}

it('Should render basic review', () => {
  const review = mount(<Review {...defaultReviewProps} />)
  assertReview(review, 'Review title', 'Review content', '05 jul - 17:39', false)
})

it('Should render review response', () => {
  const review = mount(<Review {...defaultReviewProps} isResponse />)
  assertReview(review, 'Review title', 'Review content', '05 jul - 17:39', true)
})

it('Should render basic review with href link', () => {
  const review = mount(
    <Review
      {...defaultReviewProps}
      replyLinkLabel="Reply to this review"
      replyLinkHref="/edit_review"
    />,
  )
  const replyReviewLink = review.find('.kirk-button')
  expect(replyReviewLink.exists()).toBe(true)
  expect(replyReviewLink.text()).toBe('Reply to this review')
})

it('Should render basic review with onClick link', () => {
  const review = mount(
    <Review
      {...defaultReviewProps}
      replyLinkLabel="Reply to this review"
      replyLinkOnClick={() => {}}
    />,
  )
  const replyReviewLink = review.find('.kirk-button')
  expect(replyReviewLink.exists()).toBe(true)
  expect(replyReviewLink.text()).toBe('Reply to this review')
})

it('Should render review microdata annotations and HTML', () => {
  const reviewAndResponse = (
    <Fragment>
      <Review {...defaultReviewProps} />
      <Review {...defaultReviewProps} isResponse />
    </Fragment>
  )
  const component = renderer.create(reviewAndResponse).toJSON()
  expect(component).toMatchSnapshot()
})
