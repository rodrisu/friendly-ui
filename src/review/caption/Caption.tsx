import React from 'react'

import { Button, ButtonStatus } from '../../button'
import { TextCaption } from '../../typography/caption'
import { StyledCaption } from './Caption.style'

export type CaptionProps = Readonly<{
  children: string
  isoDate: string
  href?: string | JSX.Element
  onClick?: (event: React.MouseEvent<HTMLElement>) => void
  secondaryText?: string
}>

export const Caption = ({ children, href, onClick, secondaryText, isoDate }: CaptionProps) => (
  <StyledCaption>
    <time dateTime={isoDate}>
      <TextCaption>{children}</TextCaption>
    </time>
    {secondaryText && (
      <TextCaption>
        <Button status={ButtonStatus.UNSTYLED} href={href} onClick={onClick}>
          {secondaryText}
        </Button>
      </TextCaption>
    )}
  </StyledCaption>
)
