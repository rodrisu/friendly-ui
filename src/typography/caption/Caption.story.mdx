import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks'
import { text } from '@storybook/addon-knobs'

import { color } from '../../_utils/branding'
import { TYPOGRAPHY } from '../../_utils/taxonomy'
import { TextCaption } from '.'

<Meta title={`${TYPOGRAPHY}/Caption`} />

# **Caption**

<Canvas>
  <Story name="Regular">
    <h1>
      <TextCaption isInverted={false}>
        {text('The quick\n brown fox jumps\n over the lazy\n dog')}
      </TextCaption>
    </h1>
  </Story>
</Canvas>

### isInverted

<Canvas>
  <Story name="isInverted">
    <h1 style={{ backgroundColor: color.blue }}>
      <TextCaption isInverted>The quick brown fox jumps over the lazy dog</TextCaption>
    </h1>
  </Story>
</Canvas>


## Usage

```jsx
import { TextCaption } from '@blablacar/ui-library/build/typography/caption'

<TextCaption>The quick brown fox jumps over the lazy dog</TextCaption>
```

<ArgsTable of={TextCaption} />
