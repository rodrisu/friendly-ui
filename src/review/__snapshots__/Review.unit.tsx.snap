// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render review microdata annotations and HTML 1`] = `
Array [
  .c6 {
  white-space: pre-line;
  color: #708C91;
  font-size: 13px;
  font-weight: 400;
  line-height: 16px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c5 > * {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c5 > * + * {
  text-align: right;
}

.c5 .c7 {
  display: inline;
}

.c1 .section-content {
  padding-left: 24px;
  padding-right: 24px;
}

.c1 .section-content--noHorizontalSpacing {
  padding-left: 0;
  padding-right: 0;
}

.c4 {
  white-space: pre-line;
  color: #708C91;
  font-size: 16px;
  font-weight: 400;
  line-height: 20px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 8px 0;
  word-break: break-word;
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c2 {
  white-space: pre-line;
  color: #054752;
  font-size: 18px;
  font-weight: 400;
  line-height: 20px;
}

.c0 {
  padding-top: 8px;
  padding-bottom: 8px;
  margin: 0;
}

.c0.kirk-is-review-response .kirk-review-inner {
  padding-left: 24px;
}

.c0 .kirk-review-title {
  margin: 0;
}

@media (min-width:800px) {
  .c1 .section-content {
    margin-left: auto;
    margin-right: auto;
    max-width: 662px;
  }

  .c1 .section-content.section-content--large {
    max-width: 1016px;
  }
}

<blockquote
    className="c0"
    itemScope={true}
    itemType="http://schema.org/Review"
  >
    <div
      className="kirk-review-inner"
    >
      <div
        className="c1"
        role="presentation"
      >
        <div
          className="section-content"
        >
          <h2
            className="kirk-review-title"
            itemProp="reviewRating"
            itemScope={true}
            itemType="https://schema.org/Rating"
          >
            <span
              className="c2"
              itemProp="reviewRating"
            >
              Review title
            </span>
          </h2>
        </div>
      </div>
      <div
        className="c1"
        role="presentation"
      >
        <div
          className="section-content section-content--noHorizontalSpacing"
        >
          <div
            className="c3"
            role="presentation"
          >
            <p
              className="c4"
              itemProp="reviewBody"
            >
              Review content
            </p>
          </div>
          <div
            className="c5"
          >
            <time
              dateTime="2017-08-07T14:10:40.526Z"
            >
              <span
                className="c6"
              >
                05 jul - 17:39
              </span>
            </time>
          </div>
        </div>
      </div>
    </div>
  </blockquote>,
  .c6 {
  white-space: pre-line;
  color: #708C91;
  font-size: 13px;
  font-weight: 400;
  line-height: 16px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c5 > * {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c5 > * + * {
  text-align: right;
}

.c5 .c7 {
  display: inline;
}

.c1 .section-content {
  padding-left: 24px;
  padding-right: 24px;
}

.c1 .section-content--noHorizontalSpacing {
  padding-left: 0;
  padding-right: 0;
}

.c4 {
  white-space: pre-line;
  color: #708C91;
  font-size: 16px;
  font-weight: 400;
  line-height: 20px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 8px 0;
  word-break: break-word;
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c2 {
  white-space: pre-line;
  color: #054752;
  font-size: 18px;
  font-weight: 400;
  line-height: 20px;
}

.c0 {
  padding-top: 8px;
  padding-bottom: 8px;
  margin: 0;
}

.c0.kirk-is-review-response .kirk-review-inner {
  padding-left: 24px;
}

.c0 .kirk-review-title {
  margin: 0;
}

@media (min-width:800px) {
  .c1 .section-content {
    margin-left: auto;
    margin-right: auto;
    max-width: 662px;
  }

  .c1 .section-content.section-content--large {
    max-width: 1016px;
  }
}

<blockquote
    className="c0 kirk-is-review-response"
    itemScope={true}
    itemType="http://schema.org/Review"
  >
    <div
      className="kirk-review-inner"
    >
      <div
        className="c1"
        role="presentation"
      >
        <div
          className="section-content"
        >
          <h2
            className="kirk-review-title"
          >
            <span
              className="c2"
            >
              Review title
            </span>
          </h2>
        </div>
      </div>
      <div
        className="c1"
        role="presentation"
      >
        <div
          className="section-content section-content--noHorizontalSpacing"
        >
          <div
            className="c3"
            role="presentation"
          >
            <p
              className="c4"
              itemProp="reviewBody"
            >
              Review content
            </p>
          </div>
          <div
            className="c5"
          >
            <time
              dateTime="2017-08-07T14:10:40.526Z"
            >
              <span
                className="c6"
              >
                05 jul - 17:39
              </span>
            </time>
          </div>
        </div>
      </div>
    </div>
  </blockquote>,
]
`;
